Dropping out of an SSE/supply request in the most normal way is reported as "ERROR broken pipe"

With this script (and version `*` of `HTTP::Server::Tiny`):

    use HTTP::Server::Tiny;

    my $s = supply {
        whenever Supply.interval(1) {
            emit("data: { DateTime.now }\r\n\r\n".encode);
        }
    }

    sub app(%env) {
        return 200, [
            Cache-Control => 'must-revalidate, no-cache',
            Content-Type => 'text/plain; charset=utf-8, text/event-stream'
        ],
        $s;

    }

    HTTP::Server::Tiny.new(
        port => 7787,
        max-keepalive-reqs => 10
    ).run(&app);

I can see the results in the browser (Chrome 47.0.2526.80). It looks like this:

    data: 2016-04-22T14:11:39.544433+02:00

    data: 2016-04-22T14:11:40.545878+02:00

    data: 2016-04-22T14:11:41.546105+02:00

...and values keep getting added, and it's kinda cool. Really cool, actually. Kudos.

But then I press Escape in the browser, which interrupts the loading. And I get a big fat error from the app:

    [ERROR] [24503] [6] broken pipe   in method throw at gen/moar/m-CORE.setting line 21039
      in method result at gen/moar/m-CORE.setting line 31980
      in sub await at gen/moar/m-CORE.setting line 33952
      in block  at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 328
      in block  at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 326
      in method sink-all at gen/moar/m-CORE.setting line 4177
      in method sink at gen/moar/m-CORE.setting line 12857
      in method send-response at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 323
      in method run-app at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 252
      in block  at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 219
      in method parse-body at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 167
      in method handle at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 88
      in block  at /home/masak/ours/rakudo/install/share/perl6/site/sources/41D120CC8C5AC1ADBAA2CC7E606E4F8710D84044 (HTTP::Server::Tiny) line 495
      in block  at gen/moar/m-CORE.setting line 32437
      in block  at gen/moar/m-CORE.setting line 32408
      in method protect at gen/moar/m-CORE.setting line 31547
      in block  at gen/moar/m-CORE.setting line 32408
      in block  at gen/moar/m-CORE.setting line 33648
      in method emit at gen/moar/m-CORE.setting line 33645
      in method emit at gen/moar/m-CORE.setting line 33685
      in block  at gen/moar/m-CORE.setting line 34232
      in method emit-events at gen/moar/m-CORE.setting line 3045
      in block  at gen/moar/m-CORE.setting line 3033
      in block  at gen/moar/m-CORE.setting line 3028
      in method protect at gen/moar/m-CORE.setting line 31547
      in method process at gen/moar/m-CORE.setting line 3025
      in block  at gen/moar/m-CORE.setting line 34235
      in block  at gen/moar/m-CORE.setting line 31691
      in block  at gen/moar/m-CORE.setting line 31688
      in block  at gen/moar/m-CORE.setting line 31687
      in block  at gen/moar/m-CORE.setting line 31684

Which is... kinda not cool. I mean, it's acceptable, but sub-optimal. In a sense, yes, it's a "broken pipe" because I stopped listening to the request. But in another sense, I didn't disconnect in an exceptional way, I disconnected in the most normal, expected way possible. I'd be fine with one line of application log for that, not a whole big stack trace as if something went wrong.

Thanks for implementing SSE in `HTTP::Server::Tiny` in the first place! I really like how simple the code is! Hoping the above small annoyance is quite an easy fix.

--- a/lib/HTTP/Server/Tiny.pm6
+++ b/lib/HTTP/Server/Tiny.pm6
@@ -262,6 +262,13 @@ my class HTTP::Server::Tiny::Handler {
     }
 
     method !send-response($status, $headers, $body) {
+
+        CATCH {
+            when /"broken pipe"/ {
+                debug("client close sending response");
+                return;
+            }
+        }
         debug "sending response $status";
 
         my $resp_string = "$!protocol $status {get_http_status_msg $status}\r\n";

